community/kernel/src/main/java/org/neo4j/kernel/impl/scheduler/CentralJobScheduler.java

```diff
@@ -47,15 +48,30 @@
     // fashion.
     private final ConcurrentHashMap<Group,ExecutorService> workStealingExecutors;
 
-    private final ThreadGroup topLevelGroup;
+    private final TopLevelGroup topLevelGroup;
     private final ThreadPoolManager pools;
 
     private volatile boolean started;
 
+    private static class TopLevelGroup extends ThreadGroup
+    {
+        TopLevelGroup()
+        {
+            super( "Neo4j-" + INSTANCE_COUNTER.incrementAndGet() );
+        }
+
+        public void setName( String name ) throws Exception
+        {
+            Field field = ThreadGroup.class.getDeclaredField( "name" );
+            field.setAccessible( true );
+            field.set( this, name );
```

Comment:
Well, it'd be nicer to set the name on instantiation, but that would require invasive refactoring of the DI modules.

Reply:
Yeah, exactly. It struck me as a poor trade-off, because the JobScheduler is created in the PlatformModule, which is created before any edition module. So trying to get edition-specific code in before JobScheduler creation would rub against the nice hierarchy we currently have.

Reply:
It could probably be done with an extra layer of abstraction on `JobScheduler` - a factory with a name setter, that instantiates a singleton `JobScheduler` when required. Not sure we could guarantee it didn't instantiate before the name is set. Doubt it would be worth the trouble.